### DISCUSSION TITLE
Skip Rodauth GET routes

### DISCUSSION THREAD

**Discussion by atthemiko** [janko/rodauth-rails#162] [Q&A] [answered]
Is it possible to skip certain views such as validation and logout?

For Validation:
User automatically get verified when clicking on link instead of going through the form in the account_verify views

For Logout:
User automatically log out on "/logout" instead of going through the form in the logout views

**Comment by janko** [comment]
You should be able to do it by routing those requests before the `r.rodauth` call (which routes Rodauth requests), and not returning any response, which should cause Roda to return a 404 on these requests.

```rb
# app/misc/rodauth_app.rb
class RodauthApp < Rodauth::Rails::App
  route do |r|
    r.get("verify-account") {} # should return 404
    r.get("logout") {} # should return 404

    r.rodauth
  end
end
```

I saw you asked the same question on the Rodauth repo (https://github.com/jeremyevans/rodauth/issues/263), I'm curious what Jeremy will say 🙂

**Reply by atthemiko** [reply]
It still didn't log out, just return a blank page. 

Is it possible to do the same like in Devise where you could just set config.sign_out_via = :get and sign out with "/logout" url?

**Reply by janko** [reply]
Oh, I misunderstood, I thought you're already using the POST routes, and wanted to disable the GET routes. I will post a separate answer.

**Reply by atthemiko** [reply]
Thank you!

**Comment by janko** [comment, marked as answer]
I think the simplest way would be to hide the form, and automatically submit it on page load. With Stimulus and Bootstrap, you could implement it as follows:

```erb
<!-- app/views/rodauth/verify_account.html.erb -->
<%= form_with url: rodauth.verify_account_path, method: :post, data: { turbo: false, controller: "submit" }, class: "d-none" do |form| %>
  <!-- ... -->
<% end %>
```
```js
// app/javascript/controllers/submit_controller.js
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {
  connect() {
    this.element.requestSubmit()
  }
}
```
